Test/ghost/PhantomData.dfy

```diff
@@ -0,0 +1,63 @@
+// RUN: %dafny /compile:0 /dprint:- "%s" /env:0 > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+datatype PhantomData<V> = PhantomData(ghost data: V)
+
+method ParsingGhostActualBindings()
+{
+    // Check that the TupleTypeName of the following tuples is the same
+    var tuple0 := (ghost 10:=10, 3:=3, 0:=0, 1:=1, 2:=2, 4:=4, 5:=5, 6:=6, 7:=7, 8:=8, 9:=9);
+    var tuple1 := (0, 1, 2, 3, 4, 5, 6, 7, 8, 9, ghost 10);
+    assert tuple0 == tuple1;
+}
+
+method GhostBad1()
+{
+    var x := (ghost 123); // ghost not allowed in parenthesized expression
+}
+
+method GhostBad2()
+{
+    ghost var x: (ghost int) := 123; // ghost not allowed in parenthesized type
+}
+
+method GhostBad3()
+{
+    var (ghost x) := 123; // TODO(fpoli): Is it fine to disallow this?
```

Comment:
Is `var (x) := 123` even allowed? Would that be the same as `var x := 123` ?

Reply:
It's probably disallowed, but I don't get a nice error message:

```
// Program.dfy

method Test()
{
  var (x) := 123;

  assert x == 123;
}
```

```
Program.dfy(7,8): Error: type of RHS (int) does not match type of bound variable '_#Make1'
```

Reply:
`var (x) := 123` is currently not allowed, because there is no 1-tuple. However, if we support 1-tuples where the component is ghost (which I suggest we do), then we would support `var (x) := (ghost 123)`, or more generally `var (x) := e` where `e` has type `(ghost T)` for some `T`.

Reply:
I'm going to consider `var (ghost x) := 123` a type error: the LHS is an empty tuple but the RHS an integer.

Reply:
I'm confused by `var (x) := 123`, because the right-hand side doesn't even look like a singleton tuple, it just looks like a non-tuple value. I don't see how you would enable singleton tuples since `(123)` just looks `123` nested in a parenthesis expression.

Reply:
I don't think we want to allow `var (x) := (123)`. What we want is `var (x) = (ghost 123)`, which currently parses fine.